### PR TITLE
Implement `curve-ui-kit` package to represent Figma design

### PR DESCRIPTION
**Pull request by ignromanov**
- add `curve-ui-kit`
- install `MUI` & `Storybook`
- parse Figma tokens to use in MUI
- generate MUI themes from Figma tokens
- implement Typography & Button components & stories

### CODE REVIEW

File: packages/curve-ui-kit/src/shared/ui/Typography/Typography.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Typography as MuiTypography, TypographyProps as MuiTypographyProps } from '@mui/material'
+
+export interface TypographyProps extends MuiTypographyProps {}
+
+export const Typography = ({ ...props }: TypographyProps) => {
+  return <MuiTypography {...props} />
+}
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/ui/Typography/Typography.tsx:7]:
```suggestion
export { Typography, TypographyProps } from '@mui/material'
```

File: packages/curve-ui-kit/src/shared/ui/CssBaseline/muiCssBaseline.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        font-family: 'Hubot Sans';
+        font-style: normal;
+        font-display: swap;
+        font-weight: 400;
+        src: local('Hubot Sans'), local('HubotSans-Regular'), url('fonts/Hubot-Sans.woff2') format('woff2');
+        unicodeRange: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF;
+      }
+    `,
+  }
+  return component
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/ui/CssBaseline/muiCssBaseline.ts:24]: Let's inline this

File: packages/curve-ui-kit/src/shared/ui/Button/muiButton.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        color: buttonDesktop.ghost.hover['label & icon'],
+      },
+      '&:disabled': {
+        backgroundColor: buttonDesktop.ghost.disabled.fill,
+        color: buttonDesktop.ghost.disabled['label & icon'],
+      },
+    }
+  }
+
+  const muiButton: Components['MuiButton'] = {
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/ui/Button/muiButton.ts:72]: let's inline this

File: packages/curve-ui-kit/src/shared/ui/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Button as MuiButton, ButtonProps as MuiButtonProps } from '@mui/material'
+
+export interface ButtonProps extends MuiButtonProps {
+  label: string
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/ui/Button/Button.tsx:4]: Using children is much cleaner and allows for more flexibility. E.g. using icons is impossible if you require string.
```suggestion
```

File: packages/curve-ui-kit/src/entities/themes/lib/generate-themes.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { createTheme as createMuiTheme, type Theme } from '@mui/material/styles'
+import { basicMuiTheme, type ThemeKey } from '@/shared/lib/basic-theme'
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/entities/themes/lib/generate-themes.ts:2]: `'@shared/lib'` should map to [curve-lib/src/shared/lib](https://github.com/curvefi/curve-frontend/tree/b6327360c6d88e7f7b715977011f48ce10e28470/packages/curve-lib/src/shared/lib)!

File: packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ * @param excludeKeys - An array of keys to exclude from extraction.
+ * @returns An array of extracted values.
+ */
+export function extractValues<T extends FigmaValueObject<unknown, string>>(
+  obj: T,
+  excludeKeys: string[] = [],
+): ExtractedValues<T, (typeof excludeKeys)[number]> {
+  return Object.entries(obj)
+    .filter(([key]) => !excludeKeys.includes(key))
+    .map(([, value]) => value) as ExtractedValues<T, (typeof excludeKeys)[number]>
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts:13]: Aren't the functions the same? :thinking: 
```suggestion
  return extractPairs(obj, excludeKeys).map(([k, v])=>v)
```

File: packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
+/**
+ * Extracts values from a Figma token object, excluding specified keys.
+ * @param obj - The Figma token object to extract values from.
+ * @param excludeKeys - An array of keys to exclude from extraction.
+ * @returns An array of extracted values.
+ */
+export function extractValues<T extends FigmaValueObject<unknown, string>>(
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts:7]: this function is actually unused

File: packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+ */
+export function extractPairs<T extends FigmaValueObject<unknown, string>>(
+  obj: T,
+  excludeKeys: string[] = [],
+): ExtractedKeyValuePairs<T, (typeof excludeKeys)[number]> {
+  return Object.entries(obj).reduce(
+    (acc, [key, value]) => {
+      if (!excludeKeys.includes(key)) {
+        // @ts-expect-error
+        acc[key] = value
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts:30]: Let's avoid ignoring errors, even if a cast is needed
```suggestion
        return { ...acc, [key]: value}
```

File: packages/curve-ui-kit/src/shared/api/figma-tokens/resolve/resolver.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import type { CurveFigmaDesignTokens, ScreenType } from '../tokens'
+
+export type ResolvedValue<T> = T extends { type: string; value: infer V }
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/api/figma-tokens/resolve/resolver.ts:3]: this file needs some documentation - what is this and why do we need to write complicated types

File: packages/curve-ui-kit/src/shared/api/figma-tokens/tokens/curve-figma-design.tokens.json [hunk 10]
```diff
@@ -0,0 +1,6103 @@
+{
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/api/figma-tokens/tokens/curve-figma-design.tokens.json:1]: 6kb of JSON is too much for the bundle :scream: 
can't we extend the script to extract only what we need?

File: packages/curve-ui-kit/src/shared/api/figma-tokens/validation/analyze_figma_tokens.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
+import json
+import re
+from collections import defaultdict
+
+def analyze_json_file(file_path):
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/api/figma-tokens/validation/analyze_figma_tokens.py:5]: what is this?

File: packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts (hunk 7 above)
Comment (ignromanov) [reply packages/curve-ui-kit/src/shared/api/figma-tokens/extract-figma-values.ts:7]: Yes. And we don't need it anymore. Removed.

File: packages/curve-ui-kit/src/shared/api/figma-tokens/validation/analyze_figma_tokens.py (hunk 11 above)
Comment (ignromanov) [reply packages/curve-ui-kit/src/shared/api/figma-tokens/validation/analyze_figma_tokens.py:5]: This is a script for analyzing tokens in a file, handy for checking the correctness of hyperlinks within it. I've added some documentation.

File: packages/curve-ui-kit/src/entities/themes/lib/generate-themes.ts (hunk 5 above)
Comment (ignromanov) [reply packages/curve-ui-kit/src/entities/themes/lib/generate-themes.ts:2]: The only imports from this pack will be
`entities/themes`
`shared/ui`

File: packages/curve-ui-kit/.storybook/main.ts [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  },
+
+  core: {
+    disableTelemetry: true,
+  },
+
+  docs: {},
+  typescript: {},
+
+  // TODO: wait for fix https://github.com/storybookjs/storybook/issues/12129
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/.storybook/main.ts:36]: this issue is from 2020 :dizzy_face:

File: packages/curve-ui-kit/src/shared/ui/Typography/muiTypography.ts [hunk 13]
```diff
@@ -0,0 +1,24 @@
+import type { Components } from '@mui/material/styles'
+
+export const defineMuiTypography = (): Components['MuiTypography'] => {
+  return {
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/shared/ui/Typography/muiTypography.ts:4]:
```suggestion
export const defineMuiTypography = (): Components['MuiTypography'] => ({
```

File: packages/curve-ui-kit/src/shared/api/figma-tokens/tokens/curve-figma-design.tokens.json (hunk 10 above)
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/shared/api/figma-tokens/tokens/curve-figma-design.tokens.json:1]: During building, only parts of the JSON are used? Is the bundler really that smart?

Comment (ignromanov) [reply packages/curve-ui-kit/src/shared/api/figma-tokens/tokens/curve-figma-design.tokens.json:1]: Next.js has a really strong tree shaking system and cuts off unused modules.

File: packages/curve-ui-kit/.storybook/main.ts (hunk 12 above)
Comment (ignromanov) [reply packages/curve-ui-kit/.storybook/main.ts:36]: Unfortunately, so, that's why I'm using one of the proposed solutions 😭
https://github.com/storybookjs/storybook/issues/12129#issuecomment-1486722918

**Review by amytsang — APPROVED** [review]

**Pull request merged**